src/test/java/org/junitpioneer/testkit/PioneerTestKit.java

```diff
@@ -38,12 +41,21 @@ public static ExecutionResults executeTestMethod(Class<?> testClass, String test
 	 *
 	 * @param testClass Name of the test class
 	 * @param testMethodName Name of the test method (of the given class)
-	 * @param methodParameterTypes Full qualified types names of the parameters (e.g. "java.nio.file.Path")
+	 * @param methodParameterTypes Class type(s) of for the parameters
 	 * @return The execution results
+	 * @throws IllegalArgumentException when methodParameterTypes is null
 	 */
 	public static ExecutionResults executeTestMethodWithParameterTypes(Class<?> testClass, String testMethodName,
-			String methodParameterTypes) {
-		return new ExecutionResults(testClass, testMethodName, methodParameterTypes);
+			Class<?>... methodParameterTypes) {
+
+		if (null == methodParameterTypes) {
```

Comment:
Did you consider `Objects.requireNonNull(T value, String message)`? I think it conveys this intent really nicely.

Reply:
Yes, but it throws an `NullPointerException` and not an `IllegalArgumentException`. A NPE has a total different meaning than a IAE.

Reply:
I think a `NullPointerException` conveys that a certain value can not (or should not) be `null`.

Reply:
> I think a `NullPointerException` conveys that a certain value can not (or should not) be `null`.

No it indicates that a certan value *is* NULL, according to [documentation of NullPointerException](https://docs.oracle.com/javase/7/docs/api/java/lang/NullPointerException.html) (highlight by me):

> Thrown when an application **attempts to use null in a case where an object is required**. These include:
> 
> - Calling the instance method of a null object.
> - Accessing or modifying the field of a null object.
> - Taking the length of null as if it were an array.
> - Accessing or modifying the slots of null as if it were an array.
> - Throwing null as if it were a Throwable value.
> 
> Applications should throw instances of this class to indicate other illegal uses of the null object. NullPointerException objects may be constructed by the virtual machine as if suppression were disabled and/or the stack trace was not writable.

But the [documentation of IllegalArgumentExceptiuon](https://docs.oracle.com/javase/7/docs/api/java/lang/IllegalArgumentException.html) cleary show that this exeception is to be used in cases like mine:

> Thrown to indicate that a method has been passed an illegal or inappropriate argument.

Reply:
### My interpretation:

Both exceptions are correct because 
 1) it is an illegal argument
 2) it is illegal because **_it is null but an object is required_**.

Using `Objects.requireNonNull` is already used to validate arguments inside Pioneer (`TempDirectoryExtension`).
[Its documentation says](https://docs.oracle.com/javase/8/docs/api/java/util/Objects.html#requireNonNull-T-) it was designed with this purpose in mind:
> Checks that the specified object reference is not null. This method is designed primarily for doing parameter validation in methods and constructors

Anyways, if you feel strongly about this, I'll leave it alone. It fails fast either way, so what specific exception to throw and how is up to personal preference.